core/include/kernel/notif.h

```diff
@@ -89,6 +102,30 @@ struct notif_driver {
 	SLIST_ENTRY(notif_driver) link;
 };
 
+/*
+ * struct notif_itr - Interrupt notifier: notifying normal world of an event
+ *
+ * @itr_num Interrupt number used as identifier for the interrupt event
+ * @ops Notifier unpaged callback operations or NULL if none
+ */
+struct notif_itr {
+	unsigned int itr_num;
+	const struct notif_itr_ops *ops;
+};
+
+/*
+ * struct notif_itr_ops - Interrupt notifier operations
+ *
+ * @set_mask Fasctcall callback for (un)masking the event or NULL if not used
+ * @set_state Callback for enabling/disabling the event or NULL if not used
+ * @set_wakeup Callback for enabling/disabling standby wakeup source or NULL
```

Comment:
Do you have an example of what this callback should do and when it's called?

Reply:
It is aimed to be called by non-secure world from `enable_irq_wake()` and `disable_irq_wake()` in Linux kernel (see https://elixir.bootlin.com/linux/v6.3/source/include/linux/interrupt.h#L482) or like. Such a call is relayed to OP-TEE core through [optee_itr_notif_set_wake() handler](https://github.com/linaro-swg/linux/pull/112/commits/9a724e5484e5f5440547a46afca119e433969f14#diff-e496739c0d15953f39d2accb0bf7aa442921cdb6b76b91847e4d850d0c27d371R1016).

Reply:
`struct notif_itr_ops` is intended to 'interrupt notification', a service exposed to non-secure world. 
For wake up source handled inside OP-TEE, OP-TEE drivers should use OP-TEE internal means (for example we could add a `.set_wakeup` handler field in `struct itr_chip`, once https://github.com/OP-TEE/optee_os/pull/5954 is reviewed), not this ops.